packages/xod-project/src/patch.js

```diff
@@ -1354,3 +1355,152 @@ export const validateAbstractPatch = def(
     Either.of
   )
 );
+
+export const getPatchSignature = def(
+  'getPatchSignature :: Patch -> PatchSignature',
+  R.compose(
+    R.map(
+      R.compose(
+        R.fromPairs,
+        R.map(R.converge(R.pair, [Pin.getPinOrder, Pin.getPinType]))
+      )
+    ),
+    R.groupBy(Pin.getPinDirection),
+    listPins
+  )
+);
+
+// checks that a concrete patch conforms to concrete patch specification
+export const checkSpecializationMatchesAbstraction = def(
+  'checkSpecializationMatchesAbstraction :: Patch -> Patch -> Either Error Patch',
+  (abstractPatch, checkedPatch) => {
```

Comment:
Not sure that `checkPatch` is a good name (sounds like it already checked, when it just comes as an argument). Cause function named `checkSpecializationMatchesAbstraction` and it has a `abstractPatch` as a first argument, how about to name it `specializationPatch`?

Reply:
Sounds good. Done 👍🏻